### ISSUE TITLE
Multiple s2s accept dialogs in cascade

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#13690]
### Steps to reproduce:
1. Share three folders with the same user "user1@localhost/owncloud"
2. Login as "user1"

### Expected result
Accept dialogs shown one by one (or some other method with better UX)

### Actual result
Accept dialogs are shown all at once (due to the main loop):
![s2s-multidialog](https://cloud.githubusercontent.com/assets/277525/5907243/b5986708-a59d-11e4-8dae-ad2d57ec25bc.png)

### Versions
OC 8 git master (bd888748bd9d0d9e9268447501fe35abf121083b)

Note: this is an issue with low severity.

Ideally instead of a for-loop it should use a stack and callbacks/promises in a way that the next dialog is only shown once the first one was processed.

This also means that the user has no way to close all dialogs, so it would be possible for someone to troll a user with lots of dialogs.

@schiesbn @DeepDiver1975 @jnfrmarks

**Comment by PVince81** [comment]
Having the cycle implemented properly would make it possible for the user to hit the cross icon to cancel all.

But anyway I think that in the future we'll want a different mechanism for accepting shares. (CC @owncloud/designers)

**Comment by PVince81** [comment]
If needed I can help with fixing the cycling issue and make the page refresh the list only once at the end.

**Comment by PVince81** [comment]
- [ ] only refresh the file list if in root of "All files" (shares wouldn't be imported in other subdirs)

**Comment by nickvergessen** [comment]
could also display a list with all shares and then add a check box for "accept/deny all"

**Comment by schiesbn** [comment]
For 8.1 I would love to have a sidebar entry "open remote shares" or something similar (maybe even with a counter). This view should give the user a list of open shares so that he can accept/decline them one-by-one or in groups if he marks some shares before performing a operation. On login only a yellow banner at the top should appear telling the user the new "open shares" exists.

I think this would be a great UX improvement.

**Comment by PVince81** [comment]
The sidebar idea sounds good.

CC @owncloud/designers

**Comment by jancborchardt** [comment]
This seems to be a duplicate then of the »Pending shares« page? https://github.com/owncloud/core/issues/13495

**Comment by MorrisJobke** [comment]
I would say, that this is a subtask of #13495

**Comment by PVince81** [comment]
Moving to 8.2, I also prefer the overview page approach: https://github.com/owncloud/core/issues/13495

**Comment by PVince81** [comment]
Moving to 9.0. Overview page and/or notifications is the way to go.

**Comment by nickvergessen** [comment]
> Moving to 9.0. Overview page and/or notifications is the way to go.

s/`and/or`/`and `

I fixed it for you

**Comment by PVince81** [comment]
Closing in favor of https://github.com/owncloud/core/issues/19587
